api-put-object-progress.go

```diff
@@ -142,16 +143,45 @@ func (c Client) PutObjectStreaming(bucketName, objectName string, reader io.Read
 			BucketName: bucketName,
 		}
 	}
+	// This method should return error with signature v2 minioClient.
+	if c.signature.isV2() {
+		return 0, ErrorResponse{
+			Code:       "NotImplemented",
+			Message:    "AWS streaming signature v4 is not supported with minio client initialized for AWS signature v2",
+			Key:        objectName,
+			BucketName: bucketName,
+		}
+	}
+
+	// Size of the object.
+	var size int64
+
+	// Get reader size.
+	size, err = getReaderSize(reader)
+	if err != nil {
+		return 0, err
+	}
+
+	// Check for largest object size allowed.
+	if size > int64(maxMultipartPutObjectSize) {
+		return 0, ErrEntityTooLarge(size, maxMultipartPutObjectSize, bucketName, objectName)
+	}
+
+	// If size cannot be found on a stream, it is not possible
+	// to upload using streaming signature.
+	if size < 0 {
```

Comment:
Can't we use multipart upload when we don't know the full size of the object?
- miniog-go calls NewMultipartUpload which doesn't require to know the size of the object
- minio-go tries to read at least 64MiB from io.Reader before it uploads the part.
- then CompleteMultipartUpload

We need to disable parallel uploads here, also resuming upload should be avoided which means we need to create a new upload id each time we call PutObjectStreaming.

We can do this in a separate PR since there is API breakage.